server/src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,41 @@
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { errorHandler } from '../utils';
+import { HttpMethods, Separators, StatusCode, messages } from '../constants';
+import { CustomRequest } from '../types/declaration';
+import { UUID } from 'crypto';
+import 'dotenv/config';
+
+const authorizationCallBack = async (
+  req: CustomRequest,
+  res: Response,
+  next?: NextFunction,
+): Promise<Response | undefined | void> => {
+  if (req.method === HttpMethods.OPTIONS) {
+    return next?.();
+  }
+
+  // authorization = `Bearer ${token}`
+  const token = (req.headers.Authorization as string)?.split(Separators.SPACE)[1];
+
+  if (!token) {
+    return res.status(StatusCode.UNAUTHORIZED).json({ message: messages.notAuthorized });
+  }
+
+  const decodedToken = jwt.verify(token, process.env.JWT_SECRET as string);
```

Comment:
if valid jwt token provided, decodedToken is an object of decoded data. Please, make sure that this is working correctly